### PR TITLE
split parse pedigree functionality

### PR DESCRIPTION
**Pull request by hanars**
while implementing https://github.com/broadinstitute/seqr/pull/3444 there was an edge case in the code that was nearly impossible to test, essentially because `parse_pedigree_table` was being called in 2 different places with 2 pretty different sets of arguments, but the code had error handling for all the possible argument combinations. This splits the functionality out, creating a `parse_basic_pedigree_table` helper that only works on standard pedigree formats and does not allow all the extra arguments for the sample manifest

### CODE REVIEW

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: seqr/views/apis/dataset_api_tests.py [hunk 1]
```diff
@@ -103,13 +103,17 @@ def test_add_variants_dataset(self, mock_send_email, mock_send_slack, mock_rando
...
+            'I000003_na19679': {'sampleGuids': mock.ANY},
             'I000013_na20878': {'sampleGuids': [new_sample_guid]},
         })
         self.assertSetEqual(
             set(response_json['individualsByGuid']['I000002_na19678']['sampleGuids']),
             {replaced_sample_guid, existing_old_index_sample_guid}
         )
+        self.assertSetEqual(
+            set(response_json['individualsByGuid']['I000003_na19679']['sampleGuids']),
+            {'S000153_na19679', existing_sample_guid}
```
Comment (ShifaSZ) [inline seqr/views/apis/dataset_api_tests.py:115]: I made the exact change in my branch. But it must work.

**Review by ShifaSZ — APPROVED** [review]
